File: examples/paper-shadow/src/main/java/vn/id/tozydev/paperinfra/examples/shadow/ExamplePlugin.java
package vn.id.tozydev.paperinfra.examples.shadow;

import static net.kyori.adventure.text.Component.text;
import static net.kyori.adventure.text.format.NamedTextColor.*;

import kotlin.time.Clock;
import org.bukkit.plugin.java.JavaPlugin;

public class ExamplePlugin extends JavaPlugin {
  @Override
  public void onEnable() {
    getComponentLogger().info(text("Paper Infra Example Plugin enabled!", GREEN));
    getComponentLogger()
        .info(
            text("Current time: ", GRAY)
                .append(text(Clock.System.INSTANCE.now().toString(), GOLD)));
  }

  @Override
  public void onDisable() {
    getComponentLogger().info(text("Paper Infra Example Plugin disabled!", RED));
  }
}

File: examples/paper-kotlin/src/main/kotlin/vn/id/tozydev/paperinfra/examples/kotlin/ExamplePlugin.kt
package vn.id.tozydev.paperinfra.examples.kotlin

import net.kyori.adventure.text.Component.text
import net.kyori.adventure.text.format.NamedTextColor.GOLD
import net.kyori.adventure.text.format.NamedTextColor.GRAY
import net.kyori.adventure.text.format.NamedTextColor.GREEN
import org.bukkit.plugin.java.JavaPlugin
import kotlin.time.Clock
import kotlin.time.ExperimentalTime

class ExamplePlugin : JavaPlugin() {
    @OptIn(ExperimentalTime::class)
    override fun onEnable() {
        componentLogger.info(text("Paper Infra Example Plugin is enabled!", GREEN))
        componentLogger.info(
            text("Current time: ", GRAY)
                .append(text(Clock.System.now().toString(), GOLD)),
        )
    }

    override fun onDisable() {
        slF4JLogger.info("Paper Infra Example Plugin is disabled!")
    }
}
